packages/flutter_tools/lib/src/asset.dart

```diff
@@ -621,39 +625,50 @@ class ManifestAssetBundle implements AssetBundle {
     String assetBasePath,
     List<Uri> wildcardDirectories,
     Directory projectDirectory, {
-    List<String> excludeDirs = const <String>[],
+    String? flavor,
   }) {
     final List<DeferredComponent>? components = flutterManifest.deferredComponents;
     final Map<String, Map<_Asset, List<_Asset>>> deferredComponentsAssetVariants = <String, Map<_Asset, List<_Asset>>>{};
     if (components == null) {
       return deferredComponentsAssetVariants;
     }
     for (final DeferredComponent component in components) {
-      deferredComponentsAssetVariants[component.name] = <_Asset, List<_Asset>>{};
       final _AssetDirectoryCache cache = _AssetDirectoryCache(_fileSystem);
-      for (final Uri assetUri in component.assets) {
-        if (assetUri.path.endsWith('/')) {
-          wildcardDirectories.add(assetUri);
+      final Map<_Asset, List<_Asset>> componentAssets = <_Asset, List<_Asset>>{};
+      for (final AssetsEntry assetsEntry in component.assets) {
+        if (assetsEntry.flavor != null && assetsEntry.flavor != flavor) {
```

Comment:
I suggest ignoring the case where `flavor == ''` since we default to empty string

```suggestion
        if (assetsEntry.flavor != null && flavor != '' && assetsEntry.flavor != flavor) {
```

Reply:
I've removed the defaulting, so this shouldn't be a problem anymore.